src/components/Table.js

```diff
@@ -0,0 +1,110 @@
+import React, { Component } from 'react'
+import contacts from '../contacts.json'
+import Button from './Button'
+
+class Table extends Component {
+    constructor(props) {
+        super(props)
+        this.state = {
+            allContacts: [...contacts].splice(5),
+            fiveContacts: [...contacts].splice(0, 5)
+        }
+        this.randomContact = this.randomContact.bind(this)
+        this.sortByName = this.sortByName.bind(this)
+        this.sortByPop = this.sortByPop.bind(this)
+        this.deleteContact = this.deleteContact.bind(this)//El valor de this es ignorado cuando la función es llamada con el operador new.
```

Comment:
//El valor de this es ignorado cuando la función es llamada con el operador new.

ese comentario  es erroneo, el valor de this, cuando una funcion es llamada con new se fija sobre el objeto/instancia recien creada. New es el binding mas "fuerte" entre el this y su contexto